app/helpers/document_helper.rb

```diff
@@ -78,16 +78,18 @@ def attachment_reference(attachment)
   end
 
   def attachment_thumbnail(attachment)
+    image_host = attachment.attachable.draft? ? URI(Plek.new.external_url_for("draft-origin")).host : Whitehall.public_root
```

Comment:
I think that the `attachment` variable here is an `AttachmentData` object.

In which case, it's worth noting that `attachment.attachable` is an `attr_accessor` which is `nil` unless explicitly set:

```ruby
AttachmentData.last.attachable
=> nil
```

Is that okay? I haven't looked too deeply into what sets that attribute or when, but I thought I'd point out that this change introduces the assumption that it's been set before `attachment_thumbnail` is called.

Reply:
Just double-checked the relations here between Attachment, AttachmentData, and Attachable, and I think in principle `attachable` could be null but I'm not sure if that actually makes sense in practice. That would seem to imply an attachment that isn't actually attached to anything — but if it wasn't attached to anything, we couldn't be trying to render it.

On the other hand it can't hurt to be cautious and I think if I use a safe-navigation operator it'll end up defaulting to `Whitehall.public_root` anyway.